roundup.sh

```diff
@@ -258,9 +258,11 @@ do
 
                 # exit subshell with return code of last failing command. This
                 # is needed to see the return code 253 on failed assumptions.
-                # But, only do this if the error handling is activated.
-                set -E
-                trap 'rc=$?; set +x; set -o | grep "errexit.*on" >/dev/null && exit $rc' ERR
+                # But, only do this if the error handling is activated and only if the shell supports it
+                if set -o | grep "^errtrace" >/dev/null; then
```

Comment:
The point of that line is to check whether or not errtrace is a feature that is actually supported by the shell you are using before turning it on in the next nine. Otherwise, small shell implementations like eg. Busybox will fail here if we set an option that is not supported. Therefore, we do not need to care about the current state of errtrace since we will turn it on anyways if we can.